server/src/test/java/org/elasticsearch/transport/InboundHandlerTests.java

```diff
@@ -310,7 +310,7 @@ public void test_handshake_precedes_stream_version_for_compatibility() throws Ex
         InboundMessage responseMessage = new InboundMessage(responseHeader, ReleasableBytesReference.wrap(responseContent), () -> {});
 
         TransportHandshaker.HandshakeResponse response = new TransportHandshaker.HandshakeResponse(responseMessage.openOrGetStreamInput());
-        assertThat(response.getResponseVersion()).isEqualTo(Version.V_5_10_1);
+        assertThat(response.getResponseVersion()).isEqualTo(Version.CURRENT);
```

Comment:
Probably  

`int responseHeaderSize = TcpHeader.headerSize(Version.V_5_10_1.minimumCompatibilityVersion());`

should also use CURRENT? and actually use a single varibale smth like `localVersion`

Reply:
Also, I think we should do it on master and backport to 5.10, because there we use 6.0.0 which will also eventually fail when we release 6.0.0

Reply:
Please wait for @seut to check this first.

Reply:
Please do not change this, it is intentionally not using `CURRENT` as its testing a concrete scenario/change introduced with 5.10.1. There is no need at all to change it.

Reply:
I will look into the failure.

Reply:
Or, the main intention was that we do not need to update the tests on every version bump. I'll push a fix to master and backport it.

Reply:
Thanks!

Reply:
Related fix https://github.com/crate/crate/pull/17437.